pkg/webhook/admission/servingruntime/servingruntime_webhook.go

```diff
@@ -0,0 +1,169 @@
+
+type ServingRuntimeValidator struct {
+	Client  client.Client
+	Decoder *admission.Decoder
+}
+
+func (sr *ServingRuntimeValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	servingRuntime := &v1alpha1.ServingRuntime{}
+	if err := sr.Decoder.Decode(req, servingRuntime); err != nil {
+		log.Error(err, "Failed to decode serving runtime", "name", servingRuntime.Name, "namespace", servingRuntime.Namespace)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	ExistingRuntimes := &v1alpha1.ServingRuntimeList{}
+	if err := sr.Client.List(context.TODO(), ExistingRuntimes, client.InNamespace(servingRuntime.Namespace)); err != nil {
+		log.Error(err, "Failed to get serving runtime list", "namespace", servingRuntime.Namespace)
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// Only validate for priority if the new serving runtime is not disabled
+	if servingRuntime.Spec.IsDisabled() {
+		return admission.Allowed("")
+	}
+
+	for _, runtime := range ExistingRuntimes.Items {
+		if err := validateServingRuntimePriority(&servingRuntime.Spec, &runtime.Spec, servingRuntime.Name, runtime.Name); err != nil {
+			return admission.Denied(fmt.Sprintf("%s in the servingruntimes %s and %s in namespace %s", err.Error(), runtime.Name, servingRuntime.Name, servingRuntime.Namespace))
+		}
+	}
+	return admission.Allowed("")
+}
+
+// Handle validates the incoming request
+func (csr *ClusterServingRuntimeValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	clusterServingRuntime := &v1alpha1.ClusterServingRuntime{}
+	if err := csr.Decoder.Decode(req, clusterServingRuntime); err != nil {
+		log.Error(err, "Failed to decode cluster serving runtime", "name", clusterServingRuntime.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	ExistingRuntimes := &v1alpha1.ClusterServingRuntimeList{}
+	if err := csr.Client.List(context.TODO(), ExistingRuntimes); err != nil {
+		log.Error(err, "Failed to get cluster serving runtime list")
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// Only validate for priority if the new cluster serving runtime is not disabled
+	if clusterServingRuntime.Spec.IsDisabled() {
+		return admission.Allowed("")
+	}
+
+	for _, runtime := range ExistingRuntimes.Items {
+		if err := validateServingRuntimePriority(&clusterServingRuntime.Spec, &runtime.Spec, clusterServingRuntime.Name, runtime.Name); err != nil {
+			return admission.Denied(fmt.Sprintf("%s in the clusterservingruntimes %s and %s", err.Error(), runtime.Name, clusterServingRuntime.Name))
+		}
+	}
+	return admission.Allowed("")
+}
+
+func validateServingRuntimePriority(newSpec *v1alpha1.ServingRuntimeSpec, existingSpec *v1alpha1.ServingRuntimeSpec, existingRuntimeName string, newRuntimeName string) error {
```

Comment:
Would be nice to add test cases for this validation.

Reply:
Added